### PR TITLE
[CELEBORN-160][BUILD] Spilt CI workflow

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Split CI workflow into service/spark

### Why are the changes needed?

Fast CI, prepare to enable Flink CI

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -57,11 +57,16 @@
...
     <java.version>8</java.version>
+    <scala.version>2.12.15</scala.version>
     <scala.binary.version>2.12</scala.binary.version>
     <maven.compiler.source>${java.version}</maven.compiler.source>
     <maven.compiler.target>${java.version}</maven.compiler.target>
     <maven.version>3.6.3</maven.version>
 
+    <hadoop.version>3.2.1</hadoop.version>
+    <flink.version>1.14.0</flink.version>
+    <spark.version>3.3.1</spark.version>
```
Comment (pan3793) [inline pom.xml:68]: these properties will be overwrite by `-Pspark-X.Y` or `-Pflink-X.Y`

**Comment by pan3793** [comment]
This PR should be merged after https://github.com/apache/incubator-celeborn/pull/1108

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Comment by pan3793** [comment]
Merged to main/0.2

**Pull request merged**